pandas/core/series.py

```diff
@@ -4485,9 +4490,17 @@ def to_period(self, freq=None, copy=True) -> "Series":
         if copy:
             new_values = new_values.copy()
 
+        assert isinstance(self.index, ABCDatetimeIndex)
         new_index = self.index.to_period(freq=freq)
         return self._constructor(new_values, index=new_index).__finalize__(self)
 
+    # ----------------------------------------------------------------------
+    # Add index and columns
+    index: "Index" = properties.AxisProperty(
+        axis=0, doc="The index (axis labels) of the Series."
+    )
+    generic.NDFrame._internal_names_set.add("index")
```

Comment:
shouldn't this actually be on Series._internal_names_set? (and same for Frame)

Reply:
Yeah, changed.